truss/templates/trtllm/packages/triton_client.py

```diff
@@ -139,7 +139,8 @@ async def input_generator():
                     result = result.as_numpy("text_output")
                     yield result[0].decode("utf-8")
                 else:
-                    yield json.dumps({"status": "error", "message": error.message()})
+                    status = 500 if not error.status() else error.status()
```

Comment:
non-blocking: From a separation of concerns perspective, it doesn't feel right for TritonClient to be responsible for throwing the HTTPException. Problem here is now the API for the triton client is coupled tightly with FastAPI.

Potentially consider yielding some kind of result object (that could be a result or error), and then throwing the HTTException higher up in the call stack (closer to the "FastAPI" part of the app).